shared/domains/chelonia/internals.js

```diff
@@ -774,14 +773,64 @@ export default (sbp('sbp/selectors/register', {
               return
             }
 
+            // Mark contracts that have foreign keys that have been received
+            // as dirty
+            // First, we group watched keys by key and contracts
+            const keyDict = Object.create(null)
+            targetState._volatile?.watch?.forEach(([keyName, contractID]) => {
+              if (!keyDict[keyName]) {
+                keyDict[keyName] = [contractID]
+                return
+              }
+              keyDict[keyName].push(contractID)
+            })
+            // Then, see which of those contracts need to be updated
+            // $FlowFixMe[incompatible-call]
+            const contractIdsToUpdate = Array.from(new Set(Object.entries(keyDict).flatMap(([keyName, contractIDs]) => {
+              const keyId = findKeyIdByName(targetState, keyName)
+              if (
+                // Does the key exist? (i.e., is it a current key)
+                keyId &&
+                // Is it an encryption key? (signing keys don't build up a
+                // potentially invalid state because the private key isn't
+                // required for validation; however, missing encryption keys
+                // prevent message processing)
+                targetState._vm.authorizedKeys[keyId].purpose.includes('enc') &&
+                // Is this a newly set key? (avoid re-syncing contracts that
+                // haven't been affected by the `OP_KEY_SHARE`)
+                targetState._vm.authorizedKeys[keyId]._notBeforeHeight >= newestEncryptionKeyHeight
+              ) {
+                return contractIDs
+              }
+              return []
+            })))
+            // Mark these contracts as dirty
+            contractIdsToUpdate.forEach((contractID) => {
+              const targetState = cheloniaState[contractID]
+              if (!targetState) return
```

Comment:
Hmm, under what circumstance would this happen, and does it make sense to return instead of creating the state?

Reply:
I think this shouldn't normally happen (if it does, it likely indicates stale references). However, there is no 'state' to create. If the contract is synced again, it'll have the correct state as it's being synced from scratch.